lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -217,8 +217,12 @@ public void release(String channelName) {
         @Override
         public void onMessage(ProtocolMessage msg) {
             String channelName = msg.channel;
-            Channel channel;
-            synchronized(this) { channel = channels.get(channelName); }
+            Channel channel = null;
+            synchronized(this) {
+                if (channels.containsKey(channelName)) {
```

Comment:
I think that instead of checking the existing map - it would have been clearer and more explicit to check this against DETACHED protocol message.

Reply:
This was discussed a bit on Slack (https://ably-real-time.slack.com/archives/C01EPJENRD0/p1675680904225679?thread_ts=1675427813.939969&cid=C01EPJENRD0) - this behaviour is also consistent with `ably-js` :)

Reply:
Ah thanks. Given Paddy approved this I'm happy with the approach. Just to be clear that I was worried that this might be covering much more cases than the original case where 'detached' causes an attach. So I think it would be good to answer this question: Wouldn't be there any other cases where map doesn't contain that channel and a protocol message would necessitate the creation /attachment of a new channel?

Reply:
I don't believe so